notebooks/concepts/Mosviz concept notebook.ipynb

```diff
@@ -0,0 +1,542 @@
+    "    spectra2d = [CCDData.read(fn, unit=u.adu) for fn in spectra2dfns]\n",
+    "    \n",
+    "    # there's no \"standard\" image format for deimos data, so we don't include images\n",
+    "    \n",
+    "    table = Table.read(data_directory + 'zspec.fits')\n",
+    "    \n",
+    "    mosvizapp.load_spectra(spectra)\n",
+    "    mosvizapp.load_2d_spectra(spectra2d)\n",
+    "    mosvizapp.load_table(table) "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Which they would use like:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mosapp.load_data('/path/to/deimos/data/directory', format='deimos')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And then have it load the state just like the above.  They might then optionally do their own custom:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mosapp.load_images(images)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Which is science case-specific because there's no standard format.\n",
+    "\n",
+    "Then we can encourage users to contribute their own loaders to `jdaviz` and build a library of community-provided MOS viewers, just like for `specutils` and the astropy I/O machinery."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Important UI interactions\n",
+    "\n",
+    "## Table interactions\n",
+    "\n",
+    "* When the user clicks on a table row, that should update the other views to show the spectrum from the selected row\n",
+    "* At least some columns should be editable, e.g. a \"redshift\" column or a \"spectrum quality\" column, which the user should be able to access and write out the table like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tab = mosapp.get_data('table')\n",
+    "tab['redshift']  #<- a regular Astropy table column but with whatever the current state of the table in the UI\n",
+    "tab.write('output.ecsv')  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the desktop app (but *not* the notebook, if possible) there can be a save button that does exactly the above cell."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Spectrum interactions\n",
+    "\n",
+    "* When the 1d spectrum is zoomed along the x axis, the 2d spectrum should be matched to it (and vice versa).  Y-axis of both should be independent.\n",
+    "\n",
+    "## Keyboard interactions\n",
+    "\n",
+    "* There need to be keyboard shortcuts ('{'/'}', or '<'/'>'?  Or tab/ctrl-tab?) that advance to the next spectrum.\n",
```

Comment:
This is going to require some experimentation to determine feasibility. Unlike Qt glue where we had full control over the application, here we also have to figure out how this would work/interfere with the browser, Jupyter notebook, *and* Jupyter Lab shortcuts.

Reply:
I think this is straightforward with `ipyevents`, right?

FWIW, though, it's *not* a low-priority item - science users identified this as an important use case.  At the very least knowing "easy, hard, impossible" is important.